src/CachePeer.h

```diff
@@ -227,6 +228,45 @@ class CachePeer
     void countFailure();
 };
 
+/// cache_peer configuration storage
+class CachePeers
+{
+public:
+    using CachePeerList = std::vector< std::unique_ptr<CachePeer>, PoolingAllocator< std::unique_ptr<CachePeer> > >;
+    using iterator = CachePeerList::iterator;
+    using const_iterator = CachePeerList::const_iterator;
+
+    static peer_t parseNeighborType(const char *);
+
+    /// parses a cache_peer line and stores the parsed CachePeer object
+    void parse(ConfigParser &parser);
+    /// dumps the cache peer list into the StoreEntry object
+    void dump(StoreEntry *, const char *name) const;
+    /// cleans the cache peer list
+    void clear() { cachePeers.clear(); }
+
+    iterator begin() { return cachePeers.begin(); }
+    iterator end() { return cachePeers.end(); }
+    const_iterator cbegin() const { return cachePeers.cbegin(); }
+    const_iterator cend() const { return cachePeers.cend(); }
+
+    /// the current number of CachePeer objects
+    size_t size() const { return cachePeers.size(); }
+
+    /// deletes a CachePeer object
+    void remove(CachePeer *);
+
+    const_iterator firstPing() const;
+    void advanceFirstPing();
+
+private:
+    CachePeerList cachePeers; ///< the list of parsed CachePeer objects
+    // TODO: document this field
```

Comment:
I could not document this field (and related methods) because the old first_ping variable lacked any documentation - it is unclear what its purpose was.